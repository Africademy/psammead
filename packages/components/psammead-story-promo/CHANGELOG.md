# Psammead Media Indicator Changelog

<!-- prettier-ignore -->
| Version | Description |
| ------- | ----------- |
| 0.1.5   | [PR#534](https://github.com/BBC-News/psammead/pull/XXX) Remove Timestamp padding. |
| 0.1.4   | [PR#489](https://github.com/BBC-News/psammead/pull/489) Add grid fallback. |
| 0.1.3   | [PR#498](https://github.com/bbc/psammead/pull/498) Update stories to use new input provider |
| 0.1.2   | [PR#488](https://github.com/BBC-News/psammead/pull/488) Hide story summary on device width lower than 600px. |
| 0.1.1   | [PR#474](https://github.com/BBC-News/psammead/pull/474) Update Story promo headline to use Great Primer. |
| 0.1.0   | [PR#453](https://github.com/BBC-News/psammead/pull/453) Create initial package. |
