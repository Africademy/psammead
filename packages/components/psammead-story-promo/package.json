{
  "name": "@bbc/psammead-story-promo",
  "version": "8.0.2",
  "main": "dist/index.js",
  "module": "esm/index.js",
  "sideEffects": false,
  "description": "A story promo for use on index pages",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/components/psammead-story-promo"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/components/psammead-story-promo/README.md",
  "dependencies": {
    "@bbc/gel-foundations": "^5.0.1",
    "@bbc/psammead-styles": "^6.0.1"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.9.0",
    "@emotion/styled": "^10.0.27"
  },
  "scripts": {
    "install:packages": "yarn install --focus" 
  },
  "keywords": [
    "bbc",
    "story",
    "promo"
  ]
}
