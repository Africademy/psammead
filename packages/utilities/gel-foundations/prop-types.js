module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
