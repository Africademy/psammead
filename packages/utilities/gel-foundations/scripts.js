/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
