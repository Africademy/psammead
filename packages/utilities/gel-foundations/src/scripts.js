/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
