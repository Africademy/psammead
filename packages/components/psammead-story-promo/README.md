# psammead-story-promo - [![Known Vulnerabilities](https://snyk.io/test/github/bbc/psammead/badge.svg?targetFile=packages%2Fcomponents%2Fpsammead-story-promo%2Fpackage.json)](https://snyk.io/test/github/bbc/psammead?targetFile=packages%2Fcomponents%2Fpsammead-story-promo%2Fpackage.json) [![Storybook](https://raw.githubusercontent.com/storybooks/story-promo/master/badge/badge-storybook.svg?sanitize=true)](https://bbc.github.io/psammead/?path=/story/story-promo--default) [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-story-promo.svg)](https://www.npmjs.com/package/@bbc/psammead-story-promo) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

## Description

The `StoryPromo` component is designed to be used on 'index' pages, which are pages that link to other articles/stories. It supports having an image on the left of the promo with text on the right. This text can be any collection of nodes, however typically these would be a headline, text summary and timestamp.

## Installation

`npm install @bbc/psammead-story-promo`

## StoryPromo Props

| Argument | Type   | Required | Default | Example        |
| -------- | ------ | -------- | ------- | -------------- |
| image    | node   | No       | Null    | `<img>`          |
| text     | node   | Yes      | N/A     | `<h2>Title</h2>` |

## Heading Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| Script    | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }|

## Paragraph Props

| Argument  | Type | Required | Default | Example |
| --------- | ---- | -------- | ------- | ------- |
| Script    | object | yes | latin | { canon: { groupA: { fontSize: '28', lineHeight: '32',}, groupB: { fontSize: '32', lineHeight: '36', }, groupD: { fontSize: '44', lineHeight: '48', }, }, trafalgar: { groupA: { fontSize: '20', lineHeight: '24', }, groupB: { fontSize: '24', lineHeight: '28', }, groupD: { fontSize: '32', lineHeight: '36', }, }, }|

## Usage

The typical use-case of this component is as displayed below. A Image sits on the left side of the promo with text elements on the left. These text elements are typically a headline, text summary paragraph and timestamp. The `Headline` and `Paragraph` components are provided by this package and can be imported as seen below.

```jsx
import React, { Fragment } from 'react';
import StoryPromo, { Heading, Paragraph } from '@bbc/psammead-story-promo';
import { latin } from '@bbc/gel-foundations/scripts';

const Image = (
  <img src="https://foobar.com/image.jpg" />
);

const Text = (
  <Fragment>
    <Heading script={latin}>The headline of the promo</Heading>
    <Paragraph script={latin}>The summary of the promo</Paragraph>
    <time>12 March 2019</time>
  </Fragment>
);

<StoryPromo
  image={Image}
  text={Text}
/>
```

### When to use this component

The `StoryPromo` component is designed to be used within a link element to allow the user to navigate to the story on another page.

<!-- ### When not to use this component -->

### Accessibility notes

This component uses full semantic markup for the heading and paragraph, using `h3` and `p` respectively. Other accessibility factors such as image alt text and time elements are passed in as props and aren't explicitly set in this component.

## Roadmap

- Hover and focus states will be added

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead repository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
