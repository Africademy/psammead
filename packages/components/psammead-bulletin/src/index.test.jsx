import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import { Link, LiveLabel } from '@bbc/psammead-story-promo';
import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
import Image from '@bbc/psammead-image';
import {
  Bulletin,
  BulletinHeading,
  BulletinSummary,
  BulletinCTA,
} from './index';

/* eslint-disable react/prop-types */
const BulletinComponent = ({ script, service, isLive, dir, type, ctaText }) => {
  const summaryText = 'This is the summary text';
  const headlineText = 'This is the headline';
  const imgSrc =
    'https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg';
  const bulletinLink = 'https://bbc.co.uk';
  const imgAltText = 'Iron man';

  return (
    <Bulletin>
      {type === 'video' && <Image src={imgSrc} alt={imgAltText} />}
      <BulletinHeading script={script} service={service}>
        <VisuallyHiddenText>{ctaText}, </VisuallyHiddenText>
        {isLive && (
          <LiveLabel service={service} dir={dir}>
            Live
          </LiveLabel>
        )}
        <Link href={bulletinLink}>{headlineText}</Link>
      </BulletinHeading>
      <BulletinSummary script={script} service={service}>
        {summaryText}
      </BulletinSummary>
      <BulletinCTA
        type={type}
        isLive={isLive}
        dir={dir}
        service={service}
        script={script}
      >
        {ctaText}
      </BulletinCTA>
    </Bulletin>
  );
};

describe('Bulletin', () => {
  shouldMatchSnapshot(
    'should render audio correctly',
    <BulletinComponent
      script={latin}
      service="news"
      dir="ltr"
      ctaText="Listen"
      type="audio"
    />,
  );

  shouldMatchSnapshot(
    'should render video correctly',
    <BulletinComponent
      script={latin}
      service="news"
      dir="ltr"
      ctaText="Watch"
      type="video"
    />,
  );

  shouldMatchSnapshot(
    'should render live audio correctly',
    <BulletinComponent
      script={latin}
      service="news"
      dir="ltr"
      ctaText="Listen"
      type="audio"
      isLive
    />,
  );

  shouldMatchSnapshot(
    'should render live video correctly',
    <BulletinComponent
      script={latin}
      service="news"
      dir="ltr"
      ctaText="Watch"
      type="video"
      isLive
    />,
  );
});
