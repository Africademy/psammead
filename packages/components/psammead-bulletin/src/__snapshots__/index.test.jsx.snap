// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bulletin should render audio correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c6:hover,
.c6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:visited {
  color: #6E6E73;
}

.c0 {
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
}

.c1 {
  grid-column: 1 / span 6;
}

.c3 {
  grid-column: 1 / span 6;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.25rem;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
}

.c8 {
  background-color: #222222;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #FFFFFF;
  text-align: center;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem;
  padding-bottom: 1rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding: 0.5rem 0.5rem;
}

@media (min-width:63rem) {
  .c0 {
    display: block;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 2;
  }
}

@media (max-width:37.4375rem) {
  .c1 {
    display: none;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    width: 6.125rem;
  }
}

@media (min-width:63rem) {
  .c8 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0.5rem 0;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    type="audio"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    type="audio"
  >
    <h3
      class="c4"
      type="audio"
    >
      <span
        class="c5"
      >
        Listen 
      </span>
      <a
        class="c6"
        href="https://bbc.co.uk"
      >
        This is the headline
      </a>
    </h3>
    <p
      class="c7"
      type="audio"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c8"
    >
      <span
        class="c9"
      >
        <svg
          class="c10"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </span>
      Listen
    </div>
  </div>
</div>
`;

exports[`Bulletin should render live audio correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c11 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c7 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c7:hover,
.c7:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7:visited {
  color: #6E6E73;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c0 {
  background-color: #F2F2F2;
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
}

.c1 {
  grid-column: 1 / span 6;
}

.c3 {
  grid-column: 1 / span 6;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.25rem;
}

.c10 > svg {
  color: #FFFFFF;
  fill: currentColor;
}

.c9 {
  background-color: #B80000;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #FFFFFF;
  text-align: center;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding: 0 0.5rem;
  padding-bottom: 1rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding: 0.5rem 0.5rem;
}

@media (min-width:63rem) {
  .c0 {
    display: block;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 2;
  }
}

@media (max-width:37.4375rem) {
  .c1 {
    display: none;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    width: 6.125rem;
  }
}

@media (min-width:63rem) {
  .c9 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0.5rem 0;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    type="audio"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    type="audio"
  >
    <h3
      class="c4"
      type="audio"
    >
      <span
        class="c5"
      >
        Listen LIVE 
      </span>
      <span
        aria-hidden="true"
        class="c6"
        dir="ltr"
      >
        LIVE
      </span>
      <a
        class="c7"
        href="https://bbc.co.uk"
      >
        This is the headline
      </a>
    </h3>
    <p
      class="c8"
      type="audio"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c9"
    >
      <span
        class="c10"
      >
        <svg
          class="c11"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </span>
      Listen
    </div>
  </div>
</div>
`;

exports[`Bulletin should render live video correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c11 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c7 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c7:hover,
.c7:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7:visited {
  color: #6E6E73;
}

.c6 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
  padding: 0.5rem;
}

.c1 {
  grid-column: 1 / span 6;
}

.c3 {
  grid-column: 1 / span 6;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.25rem;
}

.c10 > svg {
  color: #FFFFFF;
  fill: currentColor;
}

.c9 {
  background-color: #B80000;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #FFFFFF;
  text-align: center;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding-bottom: 1rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding: 0.5rem 0;
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:63rem) {
  .c0 {
    display: block;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 3;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    grid-column: 4 / span 3;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    width: 6.125rem;
  }
}

@media (min-width:63rem) {
  .c9 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0 0 0.5rem 0;
  }
}

@media (min-width:63rem) {
  .c4 {
    padding-top: 0.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    type="video"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    type="video"
  >
    <h3
      class="c4"
      type="video"
    >
      <span
        class="c5"
      >
        Watch LIVE 
      </span>
      <span
        aria-hidden="true"
        class="c6"
        dir="ltr"
      >
        LIVE
      </span>
      <a
        class="c7"
        href="https://bbc.co.uk"
      >
        This is the headline
      </a>
    </h3>
    <p
      class="c8"
      type="video"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c9"
    >
      <span
        class="c10"
      >
        <svg
          class="c11"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </span>
      Watch
    </div>
  </div>
</div>
`;

exports[`Bulletin should render video correctly 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c6:hover,
.c6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:visited {
  color: #6E6E73;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(6,1fr);
  grid-column-gap: 1rem;
  padding: 0.5rem;
}

.c1 {
  grid-column: 1 / span 6;
}

.c3 {
  grid-column: 1 / span 6;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.25rem;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
}

.c8 {
  background-color: #222222;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #FFFFFF;
  text-align: center;
  padding: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  margin: 0;
  padding-bottom: 1rem;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #222222;
  margin: 0;
  padding: 0.5rem 0;
}

@media (min-width:37.5rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:63rem) {
  .c0 {
    display: block;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 3;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    grid-column: 4 / span 3;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    width: 6.125rem;
  }
}

@media (min-width:63rem) {
  .c8 {
    width: 100%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0 0 0.5rem 0;
  }
}

@media (min-width:63rem) {
  .c4 {
    padding-top: 0.5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    type="video"
  >
    <img
      alt="Iron man"
      class="c2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11897/production/_106613817_999_al_.jpg"
    />
  </div>
  <div
    class="c3"
    type="video"
  >
    <h3
      class="c4"
      type="video"
    >
      <span
        class="c5"
      >
        Watch 
      </span>
      <a
        class="c6"
        href="https://bbc.co.uk"
      >
        This is the headline
      </a>
    </h3>
    <p
      class="c7"
      type="video"
    >
      This is the summary text
    </p>
    <div
      aria-hidden="true"
      class="c8"
    >
      <span
        class="c9"
      >
        <svg
          class="c10"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </span>
      Watch
    </div>
  </div>
</div>
`;
